Sunrise/ViewModels/BaseViewModel.swift

```diff
@@ -0,0 +1,36 @@
+//
+// Copyright (c) 2016 Commercetools. All rights reserved.
+//
+
+import ReactiveCocoa
+import Result
+import ObjectMapper
+import Commercetools
+
+class BaseViewModel {
+
+    // Outputs
+    let alertMessageSignal: Signal<String, NoError>
+
+    let alertMessageObserver: Observer<String, NoError>
+
+    // MARK: - Lifecycle
+
+    init() {
+        let (alertMessageSignal, alertMessageObserver) = Signal<String, NoError>.pipe()
+        self.alertMessageSignal = alertMessageSignal
+        self.alertMessageObserver = alertMessageObserver
+    }
+
+    func alertMessageForErrors(errors: [NSError]) -> String {
+        return errors.map({
+            var alertMessage = ""
+            if let failureReason = $0.userInfo[NSLocalizedFailureReasonErrorKey] as? String {
+                alertMessage += "\(failureReason) :"
+            }
+            alertMessage += $0.userInfo[NSLocalizedDescriptionKey] as? String ?? $0.localizedDescription
```

Comment:
Is there a benefit using the `userInfo` dictionary here, since `$0.localizedDescription` also uses it internally and also has a fallback, if the dictionary is empty?

https://developer.apple.com/library/mac/documentation/Cocoa/Reference/Foundation/Classes/NSError_Class/#//apple_ref/occ/instp/NSError/localizedDescription